File: GXpert/GXpert.Web/Modules/Institute/Branch/BranchColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Institute.Columns;

[ColumnsScript("Institute.Branch")]
[BasedOnRow(typeof(BranchRow), CheckNames = true)]
public class BranchColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string InstituteName { get; set; }
    public string DepartmentTitle { get; set; }
    public int BranchCode { get; set; }
    public string Description { get; set; }
   
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Institute/BranchColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BranchRow } from "./BranchRow";

export interface BranchColumns {
    Id: Column<BranchRow>;
    Title: Column<BranchRow>;
    InstituteName: Column<BranchRow>;
    DepartmentTitle: Column<BranchRow>;
    Description: Column<BranchRow>;
}

export class BranchColumns extends ColumnsBase<BranchRow> {
    static readonly columnsKey = 'Institute.Branch';
    static readonly Fields = fieldsProxy<BranchColumns>();
}